VenuApp/src/main/java/infinitycodecrew/VenuApp/controllers/AuthenticationController.java

```diff
@@ -0,0 +1,148 @@
+        model.addAttribute("title", "Register");
+        return "users/register";
+    }
+
+    @PostMapping("/register")
+    public String processRegistrationForm(@ModelAttribute @Valid dto.RegisterFormDTO registerFormDTO,
+                                          Errors errors, HttpServletRequest request,
+                                          Model model) {
+        if (errors.hasErrors()) {
+            model.addAttribute("title", "Register");
+            return "users/register";
+        }
+
+
+        User existingUser = userRepository.findByUsername(registerFormDTO.getUsername());
+
+        if (existingUser != null) {
+            errors.rejectValue("username", "username.alreadyexists", "A user with that username already exists!");
+            model.addAttribute("title", "Register");
+            return "users/register";
+        }
+
+        String password = registerFormDTO.getPassword();
+        String verifyPassword = registerFormDTO.getVerifyPassword();
+        if (!password.equals(verifyPassword)) {
+            errors.rejectValue("password", "password.mismatch", "Passwords do not match!");
+            model.addAttribute("title", "Register");
+            return "users/register";
+        }
+
+        User newUser = new User(registerFormDTO.getUsername(), registerFormDTO.getPassword());
+        userRepository.save(newUser);
+        setUserInSession(request.getSession(), newUser);
+
+        return "redirect:";
+    }
+
+    @GetMapping("/login")
+    public String displayLoginForm(Model model) {
+        model.addAttribute(new dto.LoginFormDTO());
+        model.addAttribute("title", "Log In");
+        return "users/login";
+    }
+
+    @PostMapping("/login")
+    public String processLoginForm(@ModelAttribute @Valid dto.LoginFormDTO loginFormDTO,
+                                   Errors errors, HttpServletRequest request,
+                                   Model model){
+        if (errors.hasErrors()){
+            model.addAttribute("title", "Log In");
+            return "users/login";
+        }
+
+        User theUser = userRepository.findByUsername((loginFormDTO.getUsername()));
+
+        if (theUser == null) {
+            errors.rejectValue("username", "user.invalid", "This username does not exist!");
+            model.addAttribute("title", "Log In");
+            return "users/login";
+        }
+
+        String password = loginFormDTO.getPassword();
+
+        if (!theUser.isMatchingPassword(password)) {
+            errors.rejectValue("password", "password.invalid", "Incorrect password!");
+            model.addAttribute("title", "Log In");
+            return "users/login";
+        }
+
+        setUserInSession(request.getSession(), theUser);
+
+        return "redirect:";
+    }
+
+    @GetMapping("/logout")
+    public String logout(HttpServletRequest request) {
+        request.getSession().invalidate();
+        return "redirect:users/login";
+    }
+
+}
+
```

Comment:
Great work on the code! I just noticed a lot of empty lines being added in. This makes it harder to read the code. Can we be mindful of this in the future?